### PR TITLE
feat: listbox grid layout

### PR DESCRIPTION
**Pull request by veinfors**
Changes:
- Listbox supporting grid layout.
- Refactored a bit

Recent changes:
- Fixed issue with clearing search (`abortSearch`) not refreshing rows
- Fixed few issues seen when testing Direct Query

### CODE REVIEW

File: apis/nucleus/src/components/listbox/assets/horizontal-minimum-batch-size.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export default function getHorizontalMinBatchSize({ width, columnWidth, listHeight, itemSize }) {
+  const visibleCellsCount = Math.ceil(width / columnWidth) * Math.ceil(listHeight / itemSize);
+  const minSize = visibleCellsCount * 2;
```
Comment (DanielS-Qlik) [inline apis/nucleus/src/components/listbox/assets/horizontal-minimum-batch-size.js:3]:
```suggestion
  return visibleCellsCount * 2;
```

File: apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx [hunk 2]
```diff
@@ -464,6 +487,7 @@ function RowColumn({ index, style, data }) {
             {showTick && <Tick style={iconStyles} size="small" />}
           </Grid>
         )}
+        {/* </div> */}
```
Comment (DanielS-Qlik) [inline apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx:490]:
```suggestion
```

File: apis/nucleus/src/components/listbox/components/grid-list-components/derive-options.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable import/prefer-default-export */
+const FREQUENCY_MIN_SHOW_WIDTH = 80;
+
+export const getFrequencyAllowed = ({ width, layout, frequencyMode }) => {
```
Comment (DanielS-Qlik) [inline apis/nucleus/src/components/listbox/components/grid-list-components/derive-options.js:4]:
```suggestion
const getFrequencyAllowed = ({ width, layout, frequencyMode }) => {
```
then at end of file:
```
export default getFrequencyAllowed;
```
and eslint-disable can be removed

File: apis/nucleus/src/components/listbox/assets/get-list-sizes.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+const scrollBarWidth = 10; // TODO: ignore this - instead set the styling only show on hover...
+
+export default function getListSizes({ layout, width, height, checkboxes, listCount, count, textWidth }) {
+  const { layoutOptions = {} } = layout || {};
+
+  const { layoutOrder, maxVisibleRows = {}, maxVisibleColumns, dense } = layoutOptions || {};
```
Comment (veinfors) [inline apis/nucleus/src/components/listbox/assets/get-list-sizes.js:6]:
```suggestion
  const { layoutOrder, maxVisibleRows = {}, maxVisibleColumns, dense } = layoutOptions;
```
I think it should be fine like this

File: apis/nucleus/src/components/listbox/assets/horizontal-minimum-batch-size.js (hunk 1 above)
Comment (johanlahti) [reply apis/nucleus/src/components/listbox/assets/horizontal-minimum-batch-size.js:3]: I will keep it mainly for debugging purpose.

File: apis/nucleus/src/components/listbox/components/grid-list-components/derive-options.js (hunk 3 above)
Comment (johanlahti) [reply apis/nucleus/src/components/listbox/components/grid-list-components/derive-options.js:4]: I was planning to add more functions here but might not be necessary so I should probably follow your suggestion 👍

**Review by johanlahti — APPROVED** [review]
LGTM 😄

**Pull request merged**